libraries/AP_RangeFinder/AP_RangeFinder_SITL.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <AP_HAL/AP_HAL.h>
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+
+#include "AP_RangeFinder_SITL.h"
+
+extern const AP_HAL::HAL& hal;
+
+/*
+  constructor - registers instance at top RangeFinder driver
+ */
+AP_RangeFinder_SITL::AP_RangeFinder_SITL(RangeFinder::RangeFinder_State &_state, AP_RangeFinder_Params &_params, uint8_t instance) :
+    AP_RangeFinder_Backend(_state, _params),
+    sitl(AP::sitl()),
+    _instance(instance)
+{}
+
+/*
+  update distance_cm
+ */
+void AP_RangeFinder_SITL::update(void)
+{
+    const float dist = sitl->get_rangefinder(_instance);
+
+    // negative distance means nothing is connected
```

Comment:
How were you planning on doing out-of-range-low or out-of-range-high?

"later" is a valid response; nothing in here is set in stone.

We should probably try to allow for each of the enumeration entries in `RangeFinder::Status` at some stage.

Reply:
It will set the status out of range high or low if the reading is outside the min or max range as set by the params. I don't think we need all the bells and whistles, there are better sitl rangefinders for that already, this was just the minimum viable method of getting data from the physics to the rangefinder lib.